modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -232,10 +237,12 @@ def _row_lengths(self):
             A list of row partitions lengths.
         """
         if self._row_lengths_cache is None:
-            if len(self._partitions.T) > 0:
-                self._row_lengths_cache = [
-                    obj.length() for obj in self._partitions.T[0]
-                ]
+            if len(self._partitions) > 0:
+                index, self._row_lengths_cache = self._compute_axis_labels_and_lengths(
+                    0
+                )
```

Comment:
```suggestion
                (
                    index,
                    self._row_lengths_cache
                ) = self._compute_axis_labels_and_lengths(0)
```
Is that nice for black? 😄